config/quality/checkstyle/checkstyle.xml

```diff
@@ -1,238 +1,240 @@
+        <!-- See http://checkstyle.sourceforge.net/config_misc.html -->
+        <module name="ArrayTypeStyle" />
+        <module name="AvoidEscapedUnicodeCharacters" />
+        <module name="CommentsIndentation" />
+        <module name="DescendantToken" />
+        <module name="FileContentsHolder" />
+        <module name="FinalParameters" />
+        <module name="Indentation" />
+        <module name="OuterTypeFilename" />
+        <module name="TodoComment">
+            <property name="format" value="(?i)\s+CR\s+" />
+            <property name="severity" value="info" />
+        </module>
+        <module name="TodoComment">
+            <property name="format" value="(?i)\s+TODO\s+" />
+            <property name="severity" value="info" />
+        </module>
+        <module name="TrailingComment" />
+        <module name="UncommentedMain" />
+        <module name="UpperEll" />
+
+        <!-- Modifier Checks                                    -->
+        <!-- See http://checkstyle.sourceforge.net/config_modifiers.html -->
+        <module name="ModifierOrder" />
+        <module name="RedundantModifier" />
+
         <!-- Checks for Naming Conventions.                  -->
         <!-- See http://checkstyle.sourceforge.net/config_naming.html -->
+        <module name="AbbreviationAsWordInName" />
+        <module name="AbstractClassName" />
+        <module name="CatchParameterName" />
+        <module name="ClassTypeParameterName" />
         <module name="ConstantName" />
+        <module name="InterfaceTypeParameterName" />
         <module name="LocalFinalVariableName" />
-        <module name="FinalLocalVariable">
-            <property name="tokens" value="VARIABLE_DEF,PARAMETER_DEF"/>
-        </module>
         <module name="LocalVariableName" />
         <module name="MemberName" />
-        <module name="MethodName">
-            <property name="severity" value="ignore" />
-        </module>
+        <module name="MethodName" />
+        <module name="MethodTypeParameterName" />
         <module name="PackageName" />
-        <module name="ParameterName">
-            <property name="severity" value="ignore" />
-        </module>
-        <module name="StaticVariableName">
-            <property name="severity" value="ignore" />
-        </module>
+        <module name="ParameterName" />
+        <module name="StaticVariableName" />
         <module name="TypeName" />
 
-        <!-- Checks for imports                              -->
-        <!-- See http://checkstyle.sourceforge.net/config_import.html -->
-        <module name="AvoidStarImport" />
-        <module name="IllegalImport" />
-        <!-- defaults to sun.* packages -->
-        <module name="RedundantImport" />
-        <module name="UnusedImports" />
-
+        <!-- Checks for Naming Conventions.                  -->
+        <!-- See http://checkstyle.sourceforge.net/config_naming.html -->
+        <module name="Regexp" />
```

Comment:
If we use regex, probably, but we don't use it so I don't think it's needed to define a nice set of rules for it